corehq/apps/app_manager/models.py

```diff
@@ -4859,6 +4860,13 @@ def wrap(cls, data):
                            if module.get('doc_type') != 'CareplanModule']
         self = super(Application, cls).wrap(data)
 
+        translations = data.get('translations')
+        label_dict = {lang: label.get('case.search.title')
+            for lang, label in translations.items() if label}
+        for module in self.modules:
+            if hasattr(module, 'search_config'):
+                setattr(getattr(module, 'search_config'), 'title_label', label_dict)
+
```

Comment:
What's the long-term plan with this code?  Is there going to be a data migration at some point so this block of code can be removed?

Reply:
I'm not entirely sure myself, how would a data migration make this block of code unneeded? Just curious

Reply:
@esoergel yeah, there's a migration planned for this - check out the migration & cleanup sections in the [spec](https://docs.google.com/document/d/10PVDEk4MjQH0rPwLJSe5Fgi5AEG7GI2_2tC9A1qHTOY/edit#)

@dominicochoa-dimagi The idea is that there will be a data migration that will basically run this code on every application doc that we have (that uses case search), and save them all. That'll permanently move all of the title translations from `translations` to `title_label`, so you'll no longer need this code to do it.